src/scales/scale.timebase.js

```diff
@@ -0,0 +1,102 @@
+/* global window: false */
+'use strict';
+
+var moment = require('moment');
+moment = typeof(moment) === 'function' ? moment : window.moment;
+
+module.exports = function(Chart) {
+
+	var helpers = Chart.helpers;
+	var timeHelpers = helpers.time;
+
+	Chart.TimeScaleBase = Chart.Scale.extend({
+		initialize: function() {
+			if (!moment) {
+				throw new Error('Chart.js - Moment.js could not be found! You must include it before Chart.js to use the time scale. Download at https://momentjs.com');
+			}
+
+			this.mergeTicksOptions();
+
+			Chart.Scale.prototype.initialize.call(this);
+		},
+		// Get tooltip label
+		getLabelForIndex: function(index, datasetIndex) {
+			var me = this;
+			var chartData = me.chart.data;
+			var label = chartData.labels && index < chartData.labels.length ? chartData.labels[index] : '';
+			var value = chartData.datasets[datasetIndex].data[index];
+
+			if (value !== null && typeof value === 'object') {
+				label = me.getRightValue(value);
+			}
+
+			// Format nicely
+			if (me.options.time.tooltipFormat) {
+				label = timeHelpers.parseTime(me, label).format(me.options.time.tooltipFormat);
+			}
+
+			return label;
+		},
+    // Function to format an individual tick mark
+		tickFormatFunction: function(tick, index, ticks) {
+			var formattedTick;
+			var tickClone = tick.clone();
+			var tickTimestamp = tick.valueOf();
+			var major = false;
+			var tickOpts;
+			if (this.majorUnit && this.majorDisplayFormat && tickTimestamp === tickClone.startOf(this.majorUnit).valueOf()) {
+				// format as major unit
+				formattedTick = tick.format(this.majorDisplayFormat);
+				tickOpts = this.options.ticks.major;
+				major = true;
+			} else {
+				// format as minor (base) unit
+				formattedTick = tick.format(this.displayFormat);
+				tickOpts = this.options.ticks.minor;
+			}
+
+			var callback = helpers.valueOrDefault(tickOpts.callback, tickOpts.userCallback);
+
+			if (callback) {
+				return {
+					value: callback(formattedTick, index, ticks),
+					major: major
+				};
+			}
+			return {
+				value: formattedTick,
+				major: major
+			};
+		},
+		convertTicksToLabels: function() {
+			var me = this;
+			me.ticksAsTimestamps = me.ticks;
+			me.ticks = me.ticks.map(function(tick) {
```

Comment:
Might a good to add a comment or even better, change it for a vanilla for loop and avoid doubling function calls?